### PR TITLE
Add new keyboard "miniDivide MAX"

### PR DESCRIPTION
**Pull request by takashicompany**
## Description

I designed new keyboard "miniDivide MAX".
Please review this pull request.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/takashicompany/minidivide_max/config.h
Comment (drashna) [inline keyboards/takashicompany/minidivide_max/config.h:1]: This file should be removed.

File: keyboards/takashicompany/minidivide_max/info.json [hunk 1]
```diff
@@ -0,0 +1,544 @@
...
+        ],
+        "rows": [
+            "F4",
+            "F5",
+            "F6",
+            "F7",
+            "B1"
+        ]
+    },
+    "pin_compatible": "promicro",
```
Comment (drashna) [inline keyboards/takashicompany/minidivide_max/info.json:37]:
```suggestion
```

File: keyboards/takashicompany/minidivide_max/info.json
Comment (drashna) [inline keyboards/takashicompany/minidivide_max/info.json:1]: please run `qmk format-json` on this file

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/minidivide_max/info.json (hunk 1 above)
Comment (dunk2k) [reply keyboards/takashicompany/minidivide_max/info.json:37]: `pin_compatible` defined as part of `development_board` (reference: `data/mappings/defaults.hjson`)

File: keyboards/takashicompany/minidivide_max/info.json [hunk 2]
```diff
@@ -0,0 +1,544 @@
...
+                        9,
+                        3
+                    ],
+                    "x": 8.82,
+                    "y": -2.36
+                }
+            ]
+        }
+    }
+}
```
Comment (dunk2k) [inline keyboards/takashicompany/minidivide_max/info.json:544]: 1. adjust x and y positions to match physical board layout - QMK Configurator and `qmk info` have no capability for key rotation
2. add End of File
```suggestion
            "layout": [
                {"matrix": [0, 0], "x": 0, "y": 0.4},
                {"matrix": [0, 1], "x": 1, "y": 0.4},
                {"matrix": [0, 2], "x": 2, "y": 0.2},
                {"matrix": [0, 3], "x": 3, "y": 0},
                {"matrix": [0, 4], "x": 4, "y": 0.2},
                {"matrix": [0, 5], "x": 5, "y": 0.4},
                {"matrix": [5, 0], "x": 9, "y": 0.4},
                {"matrix": [5, 1], "x": 10, "y": 0.2},
                {"matrix": [5, 2], "x": 11, "y": 0},
                {"matrix": [5, 3], "x": 12, "y": 0.2},
                {"matrix": [5, 4], "x": 13, "y": 0.4},
                {"matrix": [5, 5], "x": 14, "y": 0.4},

                {"matrix": [1, 0], "x": 0, "y": 1.4},
                {"matrix": [1, 1], "x": 1, "y": 1.4},
                {"matrix": [1, 2], "x": 2, "y": 1.2},
                {"matrix": [1, 3], "x": 3, "y": 1},
                {"matrix": [1, 4], "x": 4, "y": 1.2},
                {"matrix": [1, 5], "x": 5, "y": 1.4},
                {"matrix": [6, 0], "x": 9, "y": 1.4},
                {"matrix": [6, 1], "x": 10, "y": 1.2},
                {"matrix": [6, 2], "x": 11, "y": 1},
                {"matrix": [6, 3], "x": 12, "y": 1.2},
                {"matrix": [6, 4], "x": 13, "y": 1.4},
                {"matrix": [6, 5], "x": 14, "y": 1.4},

                {"matrix": [2, 0], "x": 0, "y": 2.4},
                {"matrix": [2, 1], "x": 1, "y": 2.4},
                {"matrix": [2, 2], "x": 2, "y": 2.2},
                {"matrix": [2, 3], "x": 3, "y": 2},
                {"matrix": [2, 4], "x": 4, "y": 2.2},
                {"matrix": [2, 5], "x": 5, "y": 2.4},
                {"matrix": [7, 0], "x": 9, "y": 2.4},
                {"matrix": [7, 1], "x": 10, "y": 2.2},
                {"matrix": [7, 2], "x": 11, "y": 2},
                {"matrix": [7, 3], "x": 12, "y": 2.2},
                {"matrix": [7, 4], "x": 13, "y": 2.4},
                {"matrix": [7, 5], "x": 14, "y": 2.4},

                {"matrix": [3, 0], "x": 0, "y": 3.4},
                {"matrix": [3, 1], "x": 1, "y": 3.4},
                {"matrix": [3, 2], "x": 2, "y": 3.2},
                {"matrix": [3, 3], "x": 3, "y": 3},
                {"matrix": [3, 4], "x": 4, "y": 3.2},
                {"matrix": [3, 5], "x": 5, "y": 3.4},
                {"matrix": [8, 0], "x": 9, "y": 3.4},
                {"matrix": [8, 1], "x": 10, "y": 3.2},
                {"matrix": [8, 2], "x": 11, "y": 3},
                {"matrix": [8, 3], "x": 12, "y": 3.2},
                {"matrix": [8, 4], "x": 13, "y": 3.4},
                {"matrix": [8, 5], "x": 14, "y": 3.4},

                {"matrix": [4, 2], "x": 3, "y": 4.1},
                {"matrix": [4, 3], "x": 4, "y": 4.3},
                {"matrix": [4, 4], "x": 5, "y": 4.5},
                {"matrix": [4, 5], "x": 6, "y": 4.7},

                {"matrix": [9, 0], "x": 8, "y": 4.7},
                {"matrix": [9, 1], "x": 9, "y": 4.5},
                {"matrix": [9, 2], "x": 10, "y": 4.3},
                {"matrix": [9, 3], "x": 11, "y": 4.1}
            ]
        }
    }
}

```

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/minidivide_max/info.json
Comment (dunk2k) [reply keyboards/takashicompany/minidivide_max/info.json:1]: bump

**Review by dunk2k — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/minidivide_max/keymaps/via/keymap.c [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    ),
+
+    [6] = LAYOUT(
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS,
+        KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS, KC_TRNS
+    )
```
Comment (drashna) [inline keyboards/takashicompany/minidivide_max/keymaps/via/keymap.c:62]: Empty layer isn't needed, as it automatically gets populated with transparencies during via init. 
```suggestion
```

**Comment by takashicompany** [comment]
@drashna Can you review a pull request?

File: keyboards/takashicompany/minidivide_max/info.json [hunk 4]
```diff
@@ -0,0 +1,156 @@
...
+    "development_board": "promicro",
+    "diode_direction": "COL2ROW",
+    "features": {
+        "bootmagic": true,
+        "command": false,
+        "console": false,
+        "extrakey": true,
+        "mousekey": true,
+        "nkro": true,
+        "oled": true,
```
Comment (waffle87) [inline keyboards/takashicompany/minidivide_max/info.json:17]: If OLEDs are enabled at the keyboard level, there needs to exist some functionality for them in a `minidivide_max.c` file using the `oled_task_kb()` function.

File: keyboards/takashicompany/minidivide_max/info.json [hunk 5]
```diff
@@ -0,0 +1,156 @@
...
+            "B4",
+            "B5"
+        ],
+        "rows": [
+            "F4",
+            "F5",
+            "F6",
+            "F7",
+            "B1"
+        ]
```
Comment (waffle87) [inline keyboards/takashicompany/minidivide_max/info.json:35]:
```suggestion
        "cols": ["D4", "C6", "D7", "E6", "B4", "B5"],
        "rows": ["F4", "F5", "F6", "F7", "B1"]
```

File: keyboards/takashicompany/minidivide_max/info.json [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+            "snake": true,
+            "static_gradient": true,
+            "twinkle": true
+        },
+        "led_count": 16,
+        "sleep": true,
+        "split_count": [
+            8,
+            8
+        ]
```
Comment (waffle87) [inline keyboards/takashicompany/minidivide_max/info.json:55]:
```suggestion
        "split_count": [8, 8]
```

File: keyboards/takashicompany/minidivide_max/info.json [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+                    "C6",
+                    "D4"
+                ],
+                "rows": [
+                    "F4",
+                    "F5",
+                    "F6",
+                    "F7",
+                    "B1"
+                ]
```
Comment (waffle87) [inline keyboards/takashicompany/minidivide_max/info.json:75]:
```suggestion
                "cols": ["B5", "B4", "E6", "D7", "C6", "D4"],
                "rows": ["F4", "F5", "F6", "F7", "B1"]
```

File: keyboards/takashicompany/minidivide_max/keymaps/via/config.h
Comment (waffle87) [inline keyboards/takashicompany/minidivide_max/keymaps/via/config.h:1]: This file should be removed.

File: keyboards/takashicompany/minidivide_max/info.json [hunk 8]
```diff
@@ -0,0 +1,156 @@
...
+    },
+    "url": "",
+    "usb": {
+        "device_version": "0.0.1",
+        "pid": "0x0054",
+        "vid": "0x7463"
+    },
+    "ws2812": {
+        "pin": "D3"
+    },
```
Comment (waffle87) [inline keyboards/takashicompany/minidivide_max/info.json:88]:
```suggestion
    },
    "dynamic_keymap": {
        "layer_count": 7
    },
```

**Review by waffle87 — CHANGES_REQUESTED** [review]

File: keyboards/takashicompany/minidivide_max/info.json (hunk 4 above)
Comment (takashicompany) [reply keyboards/takashicompany/minidivide_max/info.json]: I fixed.

File: keyboards/takashicompany/minidivide_max/keymaps/via/config.h
Comment (takashicompany) [reply keyboards/takashicompany/minidivide_max/keymaps/via/config.h:1]: I deleted.

**Review by waffle87 — APPROVED** [review]

**Review by drashna — APPROVED** [review]

**Pull request merged**